src/planscape/plan/models.py

```diff
@@ -70,46 +71,67 @@ class Project(models.Model):
     max_slope: models.FloatField = models.FloatField(null=True)
 
 
+class ConfigPriority(models.Model):
+    # TODO: migrate to Config
+    project = models.ForeignKey(Project, on_delete=models.CASCADE, null=False)
+
+    condition = models.ForeignKey(
+        'conditions.Condition', on_delete=models.CASCADE)
+
+
 class Scenario(models.Model):
     """
-    A Scenario is associated with one User, the owner, and one Project. It has optional user-specified
-    prioritization parameters.
+    A Scenario is associated with one owner, one Plan, and one Project.
+    It contains user specified fields like 'notes'.
     """
     # TODO: Change "null=True" so that owner is not nullable. Currently owner can be null because
     # we want alpha users to not be signed in.
     owner = models.ForeignKey(
         User, on_delete=models.CASCADE, null=True)  # type: ignore
-    
-    plan = models.ForeignKey(Plan, on_delete=models.CASCADE, null=True)  # type: ignore
+
+    plan = models.ForeignKey(
+        Plan, on_delete=models.CASCADE, null=False)  # type: ignore
+
+    project = models.ForeignKey(
+        Project, on_delete=models.CASCADE, null=False)  # type: ignore
 
     # The creation time of the project, automatically set when the project is created.
     creation_time: models.DateTimeField = models.DateTimeField(
         null=True, auto_now_add=True)
 
-    project = models.ForeignKey(
-        Project, on_delete=models.CASCADE, null=True)  # type: ignore
-    
     notes: models.TextField = models.TextField(null=True)
 
-    favorited: models.BooleanField = models.BooleanField(null=True, default=False)
-    
+    favorited: models.BooleanField = models.BooleanField(
+        null=True, default=False)
+
+    class ScenarioStatus(models.IntegerChoices):
+        PENDING = 0
+        PROCESSING = 1
+        PROCESSED = 2
```

Comment:
nit: instead of "processed", can this be "success" or something along the lines of "success?

I think "failed" could also processed (albeit with an error).

Reply:
Changed to SUCCESS